heareval/evaluation/task_evaluation.py

```diff
@@ -63,22 +66,98 @@ def macroauc(
+def event_based_metrics(
+    predictions,
+    targets: Dict,
+    label_vocab=pd.DataFrame,
+    tolerance=0.200,
+    onset=True,
+    offset=True,
+):
+    """
+    event-based metrics - the ground truth and system output are compared at
+    event instance level;
+    https://tut-arg.github.io/sed_eval/sound_event.html#sound-event-detection
+    """
+    # Containers of events for sed_eval
+    reference_event_list = sed_eval_event_container(targets)
+    estimated_event_list = sed_eval_event_container(predictions)
+
+    metrics = sed_eval.sound_event.EventBasedMetrics(
+        event_label_list=list(label_vocab["label"]),
+        t_collar=tolerance,
+        evaluate_onset=onset,
+        evaluate_offset=offset,
+    )
 
-available_metrics = {"top1_error": top1_error, "macroauc": macroauc}
+    for filename in predictions:
+        metrics.evaluate(
+            reference_event_list=reference_event_list.filter(filename=filename),
+            estimated_event_list=estimated_event_list.filter(filename=filename),
+        )
 
+    # This (and segment_based_metrics) return a pretty large selection of metrics. We
+    # might want to add a task_metadata option to filter these for the specific
+    # metric that we are going to use to evaluate the task.
+    overall_metrics = metrics.results_overall_metrics()
+    return overall_metrics
+
+
+def segment_based_metrics(
+    predictions, targets: Dict, label_vocab=pd.DataFrame, time_resolution=1.0
+):
+    """
+    segment-based metrics - the ground truth and system output are compared in a
+    fixed time grid; sound events are marked as active or inactive in each segment;
+    https://tut-arg.github.io/sed_eval/sound_event.html#sound-event-detection
+    """
+    # Containers of events for sed_eval
+    reference_event_list = sed_eval_event_container(targets)
+    estimated_event_list = sed_eval_event_container(predictions)
+
+    metrics = sed_eval.sound_event.SegmentBasedMetrics(
+        event_label_list=list(label_vocab["label"]), time_resolution=time_resolution
+    )
 
-def task_evaluation(task_path: Path):
+    for filename in predictions:
+        metrics.evaluate(
+            reference_event_list=reference_event_list.filter(filename=filename),
+            estimated_event_list=estimated_event_list.filter(filename=filename),
+        )
 
-    metadata = json.load(task_path.joinpath("task_metadata.json").open())
-    label_vocab = pd.read_csv(task_path.joinpath("labelvocabulary.csv"))
+    overall_metrics = metrics.results_overall_metrics()
+    return overall_metrics
 
-    embedding_type = metadata["embedding_type"]
 
-    if "evaluation" not in metadata:
-        print(f"Task {task_path.name} has no evaluation config.")
-        return
+# TODO: Add additional metrics
 
+available_metrics: Dict[str, Callable] = {
+    "top1_error": top1_error,
+    "macroauc": macroauc,
+    "event_based": event_based_metrics,
+    "onset_only_event_based": partial(event_based_metrics, offset=False),
```

Comment:
We might want to have specific values (like tolerance) depend on a config variable -- we could include an optional kwargs dict for evaluation metrics in the task_metadata to support that if we anticipate needing to set different metric parameters more easily.